numba/cuda/deviceufunc.py

```diff
@@ -0,0 +1,742 @@
+"""
+Implements custom ufunc dispatch mechanism for non-CPU devices.
+"""
+from __future__ import print_function, absolute_import
+import operator
+import warnings
+from functools import reduce
+import numpy as np
+from numba.utils import longint
+from numba.utils import IS_PY3
+from numba.npyufunc.ufuncbuilder import _BaseUFuncBuilder
+from numba import sigutils, types
+from numba.typing import signature
+from numba.npyufunc.sigparse import parse_signature
+
+if IS_PY3:
+    def _exec(codestr, glbls):
+        exec(codestr, glbls)
+else:
+    eval(compile("""
```

Comment:
Note we generally use the six library for such things:
http://pythonhosted.org/six/#syntax-compatibility